core/spec.md

```diff
@@ -4103,28 +4154,102 @@ and the following Version specific attributes:
+  - If the Resource's document is to be serialized and is not empty,
+    then either `RESOURCE` or `RESOURCEbase64` MUST be present.
+  - MUST only be used if the Resource document (bytes) are in the same
+    format as the Registry Resource entity.
+  - MUST NOT be present if `RESOURCEbase64` is also present.
+  - MUST NOT be present if the Resource's `hasdocument` model attribute is
+    set to `false.
+
+##### `RESOURCEbase64` Attribute
+- Type: String
+- Description: This attribute is a base64 serialization of the corresponding
+  Resource document's contents. If the Resource document (which are stored as
+  an array of bytes) are not conformant with the format being used to serialize
+  with the Resource object (i.e. as a JSON value), then this attribute MUST be
+  used in instead of the `RESOURCE` attribute.
+- Constraints:
+  - MUST NOT be present when the Resource's Registry metadata are being
+    serialized as HTTP headers.
+  - If the Resource's document is to be serialized and it is not empty,
+    then either `RESOURCE` or `RESOURCEbase64` MUST be present.
+  - MUST be a base64 encoded string of the Resource's document.
+  - MUST NOT be present if `RESOURCE` is also present.
+  - MUST NOT be present if the Resource's `hasdocument` model attribute is
+    set to `false.
+
 #### Version IDs
 
-If a server does not support client-side specification of the `id` of a new
-Version (see the `setversionid` attribute in the [Registry
-Model](#registry-model), or if a client choose to not specify the `id`, then
-the server MUST assign new Version an `id` that is unique within the scope of
-its owning Resource.
-
-Servers MAY have their own algorithm for creation of new Version `id` values,
-but the default algorithm is as follows:
-- `id` MUST be a string serialization of a monotonically increasing (by `1`)
-  unsigned integer starting with `1` and is scoped to the owning Resource.
-- Each time a new `id` is needed, if an existing Version already has that
-  `id` then the server MUST generate the next `id` value and try again.
+If a server does not support client-side specification of the `versionid` of a
+new Version (see the `setversionid` attribute in the [Registry
+Model](#registry-model), or if a client choose to not specify the `versionid`,
+then the server MUST assign new Version an `versionid` that is unique within
+the scope of its owning Resource.
+
+Servers MAY have their own algorithm for creation of new Version `versionid`
+values, but the default algorithm is as follows:
+- `versionid` MUST be a string serialization of a monotonically increasing
+  (by `1`) unsigned integer starting with `1` and is scoped to the owning
+  Resource.
+- Each time a new `versionid` is needed, if an existing Version already has
+  that `versionid` then the server MUST generate the next `versionid` value
+  and try again.
 - The search for the next value does not restart with `1` each time, it MUST
   continue from the previously generated value.
```

Comment:
This is even true if the version with the previously generated value got deleted again, correct? By default, the "re-use" of version ids should be avoided.

Reply:
yes. I added the word "highest" see if that helps clarify it